velox/functions/prestosql/Repeat.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+class RepeatFunction : public exec::VectorFunction {
+ public:
+  static constexpr int32_t kMaxResultEntries = 10'000;
+
+  bool isDefaultNullBehavior() const override {
+    // repeat(null, n) returns an array of n nulls.
+    return false;
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    const auto numRows = rows.size();
+    auto constantCount = 0;
+    auto totalCount = 0;
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    DecodedVector* countDecoded = nullptr;
+
+    auto pool = context.pool();
+    BufferPtr nulls;
+    uint64_t* rawNulls = nullptr;
+    if (args[1]->isConstantEncoding()) {
+      if (args[1]->as<ConstantVector<int32_t>>()->isNullAt(0)) {
+        // If count is a null constant, the result should be all nulls.
+        context.moveOrCopyResult(
+            BaseVector::createNullConstant(outputType, numRows, pool),
+            rows,
+            result);
+        return;
+      }
+      constantCount = args[1]->as<ConstantVector<int32_t>>()->valueAt(0);
+      checkCount(constantCount);
```

Comment:
Will an error throws here be suppressed correctly if the expression is wrapped in a TRY? Do we have tests for TRY and constant 'count'?

Reply:
Good idea. Let me add some tests.